axlearn/common/input_ranking.py

```diff
@@ -0,0 +1,54 @@
+# Copyright © 2023 Apple Inc.
+
+"""Input processing utilities on tf.data for ranking-related tasks."""
+from typing import Dict
+
+import seqio
+import tensorflow as tf
+
+from axlearn.common import input_tf_data
+
+
+def rank_by_value(
+    *, input_key: str, output_key: str, ascending: bool = True, allow_ties: bool = False
+) -> input_tf_data.DatasetToDatasetFn:
+    """Returns a DatasetToDatasetFn that stores the ranks of input_field in output_field.
+
+    Note the rank starts at 1.
+
+    Args:
+        input_key: The field whose value will be ranked.
+        output_key: The field to store the ranks into.
+        ascending: True to rank in ascending order or False to rank in descending order.
+        allow_ties: If true, multiple elements could have the same rank. Ranks could have gaps
```

Comment:
Comment on how we break ties when allow_ties=False.

Reply:
Done.